specutils/spectra/spectrum1d.py

```diff
@@ -85,8 +85,9 @@ def __init__(self, flux=None, spectral_axis=None, wcs=None,
         # In the case where the arithmetic operation is being performed with
         # a single float, int, or array object, just go ahead and ignore wcs
         # requirements
-        if isinstance(flux, float) or isinstance(flux, int) \
-                or not isinstance(flux, u.Quantity):
+        if np.ndim(flux) == 0 and (isinstance(flux, float)
+            or isinstance(flux, int)
+            or not isinstance(flux, u.Quantity)):
```

Comment:
```suggestion
        if not isinstance(flux, u.Quantity) and (np.ndim(flux) == 0 and
                                                (isinstance(flux, float)
                                                 or isinstance(flux, int))):
```
A minor adjustment, but I think it's probably better to *first* check the `Quantity` bit since that's a lot cheaper than `np.ndim` (which potentially does some conversions)